src/main/java/gyro/google/cloudsql/DatabaseInstanceResource.java

```diff
@@ -0,0 +1,950 @@
+            config.copyFrom(model.getReplicationCluster());
+            setReplicationCluster(config);
+        }
+
+        setScheduledMaintenance(null);
+        if (model.getScheduledMaintenance() != null) {
+            DbSqlScheduledMaintenance config = new DbSqlScheduledMaintenance();
+            config.copyFrom(model.getScheduledMaintenance());
+            setScheduledMaintenance(config);
+        }
+
+        setSettings(null);
+        if (model.getSettings() != null) {
+            DbSettings config = new DbSettings();
+            config.copyFrom(model.getSettings());
+            setSettings(config);
+        }
+
+        setSuspensionReason(null);
+        if (model.getSuspensionReason() != null && !model.getSuspensionReason().isEmpty()) {
+            setSuspensionReason(model.getSuspensionReason());
+        }
+
+        setServerCaCert(null);
+        if (model.getServerCaCert() != null) {
+            DbSslCert config = new DbSslCert();
+            config.copyFrom(model.getServerCaCert());
+            setServerCaCert(config);
+        }
+
+        setUpgradableDatabaseVersions(null);
+        if (model.getUpgradableDatabaseVersions() != null) {
+            setUpgradableDatabaseVersions(model.getUpgradableDatabaseVersions().stream()
+                .map(upgradableDatabaseVersion -> {
+                    DbAvailableDatabaseVersion config = new DbAvailableDatabaseVersion();
+                    config.copyFrom(upgradableDatabaseVersion);
+                    return config;
+                })
+                .collect(java.util.stream.Collectors.toList()));
+        }
+
+        setGeminiConfig(null);
+        if (model.getGeminiConfig() != null) {
+            DbGeminiInstanceConfig config = new DbGeminiInstanceConfig();
+            config.copyFrom(model.getGeminiConfig());
+            setGeminiConfig(config);
+        }
+
+        setOutOfDiskReport(null);
+        if (model.getOutOfDiskReport() != null) {
+            DbSqlOutOfDiskReport config = new DbSqlOutOfDiskReport();
+            config.copyFrom(model.getOutOfDiskReport());
+            setOutOfDiskReport(config);
+        }
+
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        SQLAdmin client = createClient(SQLAdmin.class);
+        DatabaseInstance instance = client.instances().get(getProjectId(), getName()).execute();
+
+        if (instance == null) {
+            return false;
+        }
+
+        copyFrom(instance);
+
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        SQLAdmin client = createClient(SQLAdmin.class);
+        DatabaseInstance databaseInstance = new DatabaseInstance();
+
+        if (getReplicationCluster() != null) {
+            throw new GyroException(
+                "The 'replication-cluster' field is not supported for creating a new instance. It can only be configured in an update call.");
```

Comment:
Rather than throwing the error, we should follow up with an update call to handle this, rather than making the user do gyro up again

Reply:
So that problem here is that there is a sort of circular dependency. To set this value, a replica of the primary instance (which is basically another `DatabaseInstanceResource`) needs to be created first which then needs to be added back to the primary instance.
I don't think an update call would solve the issue here unfortunately.